File: test/rails_app/app/components/server_caller.rb
class ServerCaller < Netzke::Base
  action :bug_server # Actual action's text is set in en.yml

  js_properties(
    :title => "Server Caller",
    :html => "Wow",
    :bbar => [:bug_server.action]
  )

  js_method :on_bug_server, <<-JS
    function(){
      this.whatsUp();
      this.update('You should see the response from the server in the title bar the very next moment');
    }
  JS

  endpoint :whats_up do |params|
    {:set_title => "All quiet here on the server"}
  end

end

File: test/core_test_app/app/components/server_caller/javascripts/server_caller.js
{
  onBugServer: function(){
    this.whatsUp();
    this.update('You should see the response from the server in the title bar the very next moment');
  },

  onNoResponse: function(){
    this.noResponse({}, function(){
      this.update('Successfully called endpoint with no response (this is a callback)');
    }, this);
  },

  onMultipleArguments: function(){
    this.multipleArguments();
  },

  takeTwoArguments: function(first, second){
    this.update("Called a function with two arguments: " + first + ", " + second);
  },

  onArrayAsArgument: function() {
    this.arrayAsArgument();
  },

  takeArrayAsArgument: function(arry) {
    var arryAsString = "['"+ arry.join("', '") + "']";
    this.update("Called a function with array as arguments: " + arryAsString);
  },

  onCallWithGenericCallbackAndScope: function(){
    var that=this;
    var fancyScope={
      setFancyTitle: function () {
        that.setTitle("Fancy title" + " set!");
      }
    };
    this.doNothing({}, function () {
      this.setFancyTitle();
    }, fancyScope);
  },

}
